modules/hitlnext/src/views/full/Skill.tsx

```diff
@@ -0,0 +1,49 @@
+import { FormGroup, NumericInput, Checkbox } from '@blueprintjs/core'
+
+import _ from 'lodash'
+import React, { useState, FC, useEffect } from 'react'
+import { SkillData, SkillProps } from '../../types'
+
+const Skill: FC<SkillProps<SkillData>> = ({ onDataChanged, onValidChanged, initialData }) => {
+  const [timeoutDelay, setTimeoutDelay] = useState(0)
+  const [redirectNoAgent, setRedirectNoAgent] = useState(false)
+
+  useEffect(() => {
+    if (_.isEmpty(initialData)) {
+      return
+    }
+
+    setTimeoutDelay(initialData.timeoutDelay)
+    setRedirectNoAgent(initialData.redirectNoAgent)
+  }, [])
+
+  useEffect(() => {
+    onDataChanged({
+      timeoutDelay,
+      redirectNoAgent
+    })
+
+    onValidChanged(true)
+  }, [timeoutDelay, redirectNoAgent])
+
+  return (
+    <div>
+      <FormGroup
+        label="Delay before the user times out waiting for an agent (in seconds)"
+        helperText="Once this delay is exceeded, the agent request is aborted (0 = never timeouts)"
```

Comment:
Can you add that it needs to be set to the same value on the action that creates the handoff? I don't think this is really intuitive. Or at least you might not know it without looking at the code.

Reply:
You can either use the action or the skill. Why? backward compatibility

Reply:
It wasn't clear to me that you had to either use the skill or the action not both.